src/code/code_800F7260.c

```diff
@@ -1,40 +1,639 @@
+u8 D_80133340[4] = {
+    0x53,
+    0x45,
+    0x00,
+    0x00,
+};
 
+char D_80133344[] = VT_COL(RED, WHITE) "<INAGAKI CHECK> dist over! flag:%04X ptr:%08X pos:%f-%f-%f" VT_RST "\n";
+
+s32 D_8013338C = 0;
+
+UNK_TYPE D_80133390[] = {
+    0x53455120,
+    0x48000000,
+};
+
+UNK_TYPE D_80133398[] = {
+    0x20202020,
+    0x4C000000,
+};
+
+u8 D_801333A0 = 0;
+
+u8 D_801333A4 = 0;
+
+/**
+ * Array of pointers to arrays of SoundBankEntry of sizes: 9, 12, 22, 20, 8, 3, 5
+ *
+ * 0 : Player Bank          size 9
+ * 1 : Item Bank            size 12
+ * 2 : Environment Bank     size 22
+ * 3 : Enemy Bank           size 20
+ * 4 : System Bank          size 8
+ * 5 : Ocarina Bank         size 3
+ * 6 : Voice Bank           size 5
+ */
+SoundBankEntry* gSoundBanks[7] = {
+    D_8016BAD0, D_8016BC80, D_8016BEC0, D_8016C2E0, D_8016C6A0, D_8016C820, D_8016C8B0,
+};
+
+u8 sBankSizes[ARRAY_COUNT(gSoundBanks)] = {
+    9, 12, 22, 20, 8, 3, 5,
+};
+
+u8 D_801333CC = 0;
+
+u16 D_801333D0 = 0;
+
+Vec3f D_801333D4 = { 0.0f, 0.0f, 0.0f };
+
+f32 D_801333E0 = 1.0f;
+
+UNK_TYPE D_801333E4 = 0;
+
+f32 D_801333E8 = 0.0f;
+
+UNK_TYPE D_801333EC = 0;
+
+u8 D_801333F0 = 0;
+
+u8 D_801333F4 = 0;
+
+u8 D_801333F8 = 0;
+
+extern Struct_800F738C D_8016C9A0[];
+extern Struct_800F7CEC D_8016E1B8[7][3]; // total size = 0xA8
+extern Struct_800F8EA0 D_8016E270[];
+extern Struct_8013331C* D_8013331C[9];
+
+void func_800F7260(u16 arg0) {
+    u8 bankId;
+
+    for (bankId = 0; bankId < ARRAY_COUNT(gSoundBanks); bankId++) {
+        if (arg0 & 1) {
+            D_8016E264[bankId] = 1;
+        } else {
+            D_8016E264[bankId] = 0;
+        }
+        arg0 = (arg0 >> 1) & 0xFFFF;
+    }
+}
+
+void func_800F72B8(u8 arg0) {
+    D_801333D0 |= (1 << arg0);
+    func_800FA240(0, 2, 0x40, 0xF);
+    func_800FA240(3, 2, 0x40, 0xF);
+}
+
+void func_800F731C(u8 arg0) {
+    D_801333D0 &= ((1 << arg0) ^ 0xFFFF);
+    if (D_801333D0 == 0) {
+        func_800FA240(0, 2, 0x7F, 0xF);
+        func_800FA240(3, 2, 0x7F, 0xF);
+    }
+}
+
+void Audio_PlaySoundGeneral(u16 sfxId, Vec3f* a1, u8 a2, u32* a3, u32* a4, u32* a5) {
+    u8 i;
+    Struct_800F738C* phi_v0;
+
+    if (D_8016E264[SFX_BANK_SHIFT(sfxId)] == 0) {
+        phi_v0 = &D_8016C9A0[D_801333A0];
+        if (D_801333F4 == 0) {
+            for (i = 0; i < 10; i++) {
+                if (sfxId == D_8016E2E0[i]) {
+                    if (D_8016E310[i] == 0) {
+                        sfxId = D_8016E2F8[i];
+                    } else {
+                        phi_v0->sfxId = D_8016E2F8[i];
+                        phi_v0->pos = a1;
+                        phi_v0->unk_8 = a2;
+                        phi_v0->unk_C = a3;
+                        phi_v0->unk_10 = a4;
+                        phi_v0->unk_14 = a5;
+                        D_801333A0++;
+                        phi_v0 = &D_8016C9A0[D_801333A0];
+                    }
+                    i = 10;
```

Comment:
Interesting this was the method of breaking out of the loop

Reply:
sm64 audio has a bunch of these reimplemented breaks too, seems like something to watch out for in other audio files as well

Reply:
Does it actually set i to 10? Or just breaks?

Reply:
It does yeah, it sets it to 10 then increments the loop counter as if it were about to continue, only stopping since it doesn't satisfy the condition to continue the loop. Replacing it with a proper `break` ruins codegen all over the function.

Reply:
I'd like to know what was going through their heads when they decided that